airbyte-scheduler/client/src/main/java/io/airbyte/scheduler/client/DefaultSynchronousSchedulerClient.java

```diff
@@ -106,18 +117,20 @@ public SynchronousResponse<ConnectorSpecification> createGetSpecJob(final String
 
     return execute(
         ConfigType.GET_SPEC,
+        jobSpecConfig,
         null,
         jobId -> temporalClient.submitGetSpec(UUID.randomUUID(), 0, jobSpecConfig),
         ConnectorJobOutput::getSpec,
         null);
   }
 
   @VisibleForTesting
-  <T, U> SynchronousResponse<T> execute(final ConfigType configType,
-                                        @Nullable final UUID connectorDefinitionId,
-                                        final Function<UUID, TemporalResponse<U>> executor,
-                                        final Function<U, T> outputMapper,
-                                        final UUID workspaceId) {
+  <T, U, V> SynchronousResponse<T> execute(final ConfigType configType,
+                                           final V jobConfig,
```

Comment:
@benmoriceau, do you know if there's a risk of leaking secrets? I am not sure where this happens compared to the secret expansion.

Reply:
Somehow my previous comment didn't get publish.

I think that Jimmy is right. We need to call `prepareSecretsForOutput` from the class `JsonSecretsProcessor` before reporting the job config. I wish we can make that automated... 

That's a great catch Jimmy!

Reply:
It is impossible to say if it has been sanitize before so it is preferable to sanitize it as close as possible to the print/export.

Reply:
Interesting! I didn't realize these job configs would contain secrets. However, the configs aren't being sent to the JobErrorReportingClient - we're just using them to get the docker image like so:

https://github.com/airbytehq/airbyte/blob/da903d43e09f4e97e74a441d6eedf394de0532f2/airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/JobErrorReporter.java#L118

If this is still a concern and we don't want the job configs to go anywhere near the JobErrorReporter, then I could change this to instead pass the docker image directly instead of the whole config, since masking secrets requires the connector spec, which I don't know if we have access to.

Reply:
Even if the current goal is to only get the docker image, it increases the risk since having the object around will be tempting for someone to dump it for more details. I'd be fine passing a sanitized version of the config.

I expect that we will want to add more information through this pipe as time goes. Looking at other comment, `workflowId` would also be a good candidate.
That brings me back to my other question of passing a POJO as a container for debugging/error reporting details.
For example, what about a `record JobDetails(UUID connectionId, UUID workflowId, String dockerImage)` that we could extend later on in order to avoid having to change all the function signatures every time?

Reply:
Nice @gosusnp ! I have implemented this in [833ddcb](https://github.com/airbytehq/airbyte/pull/14818/commits/833ddcb7eaa83ac715b2f109261875fbf7e99d1a), though I had to implement a different POJO for sync jobs as it required something slightly different (e.g. two docker images) here [0de45f8](https://github.com/airbytehq/airbyte/pull/14818/commits/0de45f8448a709bd49c4e6b934ca057c21eb6349). I think this should be more future proof and addresses the concerns about secrets in configs.